pkg/blockbuilder/blockbuilder_test.go

```diff
@@ -215,14 +208,14 @@ func TestBlockBuilder_StartWithLookbackOnNoCommit(t *testing.T) {
 	})
 
 	// Nothing is consumed due to zero lag.
-	require.Eventually(t, func() bool { return kafkaCommits.Load() == 0 }, 5*time.Second, 100*time.Millisecond, "expected skipping all records before lookback period")
-
-	require.NoError(t, promtest.GatherAndCompare(reg, strings.NewReader(`
-		# HELP cortex_blockbuilder_consumer_lag_records The per-topic-partition number of records, instance needs to work through each cycle.
-		# TYPE cortex_blockbuilder_consumer_lag_records gauge
-		cortex_blockbuilder_consumer_lag_records{partition="0"} 0
-		cortex_blockbuilder_consumer_lag_records{partition="1"} 0
-	`), "cortex_blockbuilder_consumer_lag_records"))
+	require.Eventually(t, func() bool {
+		return assert.NoError(t, promtest.GatherAndCompare(reg, strings.NewReader(`
+			# HELP cortex_blockbuilder_consumer_lag_records The per-topic-partition number of records, instance needs to work through each cycle.
+			# TYPE cortex_blockbuilder_consumer_lag_records gauge
+			cortex_blockbuilder_consumer_lag_records{partition="0"} 0
+			cortex_blockbuilder_consumer_lag_records{partition="1"} 0
+		`), "cortex_blockbuilder_consumer_lag_records"))
+	}, 30*time.Second, 100*time.Millisecond)
```

Comment:
Note: here (and other similar changes), it's fine to wait that long. These assertions are terminal, so the test should not process if the outcome doesn't "eventually" happen.